components/experimental/src/dimension/percent/format.rs

```diff
@@ -0,0 +1,118 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use fixed_decimal::FixedDecimal;
+
+use writeable::Writeable;
+
+use crate::dimension::provider::percent::PercentEssentialsV1;
+
+pub struct FormattedPercent<'l> {
+    pub(crate) value: &'l FixedDecimal,
+    pub(crate) essential: &'l PercentEssentialsV1<'l>,
+}
+
+impl<'l> Writeable for FormattedPercent<'l> {
+    fn write_to<W>(&self, sink: &mut W) -> core::result::Result<(), core::fmt::Error>
+    where
+        W: core::fmt::Write + ?Sized,
+    {
+        // Construct the percent sign value.
+        let percent_sign_value = format!(
```

Comment:
This is close but it won't build because `format!` is in the prelude only in `std`, but this should work with `no_std`.

Since you did your work last winter, we now have the `icu_pattern` utility which does this stuff better. It will make the smallest, cleanest, fastest code if you change `PercentEssentialsV1` to use `icu_pattern`.